akka-persistence-typed/src/test/java/akka/persistence/typed/javadsl/PersistentActorCompileOnlyTest.java

```diff
@@ -0,0 +1,251 @@
+      @Override
+      public ExampleState initialState() {
+        return new ExampleState();
+      }
+
+      @Override
+      public CommandHandler<MyCommand, MyEvent, ExampleState> commandHandler() {
+        return commandHandlerBuilder(MyCommand.class)
+          .matchCommand(Cmd.class, (ctx, state, cmd) -> Effect().persist(new Evt(cmd.data))
+            .andThen(() -> cmd.sender.tell(new Ack())))
+          .build();
+      }
+
+      @Override
+      public EventHandler<MyEvent, ExampleState> eventHandler() {
+        return eventHandlerBuilder(MyEvent.class)
+          .matchEvent(Evt.class, (state, event) -> {
+            state.events.add(event.data);
+            return state;
+          })
+          .build();
+      }
+    };
+  }
+
+  static abstract class RecoveryComplete {
+    interface Command {
+    }
+    static class DoSideEffect implements Command {
+      final String data;
+
+      DoSideEffect(String data) {
+        this.data = data;
+      }
+    }
+
+    static class AcknowledgeSideEffect implements Command {
+      final int correlationId;
+
+      AcknowledgeSideEffect(int correlationId) {
+        this.correlationId = correlationId;
+      }
+    }
+
+    interface Event {
+    }
+
+    static class IntentRecord implements Event {
+      final int correlationId;
+      final String data;
+
+      IntentRecord(int correlationId, String data) {
+        this.correlationId = correlationId;
+        this.data = data;
+      }
+    }
+
+    static class SideEffectAcknowledged implements Event {
+      final int correlationId;
+
+      SideEffectAcknowledged(int correlationId) {
+        this.correlationId = correlationId;
+      }
+    }
+
+    static class EventsInFlight {
+      final int nextCorrelationId;
+      final Map<Integer, String> dataByCorrelationId;
+
+      EventsInFlight(int nextCorrelationId, Map<Integer, String> dataByCorrelationId) {
+        this.nextCorrelationId = nextCorrelationId;
+        this.dataByCorrelationId = dataByCorrelationId;
+      }
+    }
+
+    static class Request {
+      final int correlationId;
+      final String data;
+      final ActorRef<Response> sender;
+
+      Request(int correlationId, String data, ActorRef<Response> sender) {
+        this.correlationId = correlationId;
+        this.data = data;
+        this.sender = sender;
+      }
+    }
+
+    static class Response {
+      final int correlationId;
+
+      Response(int correlationId) {
+        this.correlationId = correlationId;
+      }
+    }
+
+    static ActorRef<Request> sideEffectProcessor = new TestInbox<Request>().ref();
+    static Timeout timeout = new Timeout(1, TimeUnit.SECONDS);
+    static Scheduler scheduler = null;
+
+    private static void performSideEffect(ActorRef<AcknowledgeSideEffect> sender, int correlationId, String data) {
+      CompletionStage<Response> what = ask(sideEffectProcessor, (ActorRef<Response> ar) -> new Request(correlationId, data, ar), timeout, scheduler);
```

Comment:
if this is to be included in docs the `Scheduler` must be passed in as param, from `ctx.system.scheduler`